server/service/src/invoice/outbound_return/update_lines/generate.rs

```diff
@@ -0,0 +1,92 @@
+use repository::{
+    EqualFilter, InvoiceLineFilter, InvoiceLineRepository, RepositoryError, StorageConnection,
+};
+
+use crate::invoice_line::{
+    stock_out_line::{DeleteStockOutLine, InsertStockOutLine, StockOutType, UpdateStockOutLine},
+    update_return_reason_id::UpdateLineReturnReason,
+};
+
+use super::UpdateOutboundReturnLines;
+
+pub struct GenerateResult {
+    pub lines_to_add: Vec<InsertStockOutLine>,
+    pub lines_to_update: Vec<UpdateStockOutLine>,
+    pub lines_to_delete: Vec<DeleteStockOutLine>,
+    pub update_line_return_reasons: Vec<UpdateLineReturnReason>,
+}
+
+pub fn generate(
+    connection: &StorageConnection,
+    UpdateOutboundReturnLines {
+        outbound_return_id,
+        outbound_return_lines,
+    }: UpdateOutboundReturnLines,
+) -> Result<GenerateResult, RepositoryError> {
+    let existing_lines = InvoiceLineRepository::new(connection).query_by_filter(
+        InvoiceLineFilter::new().invoice_id(EqualFilter::equal_to(&outbound_return_id)),
+    )?;
+    let check_already_exists = |id: &str| {
+        existing_lines
+            .iter()
+            .find(|line| line.invoice_line_row.id == id)
+            .is_some()
+    };
```

Comment:
Should i pull this into a separate fn? Only wanted to do the query once, rather than for every line, but this doesn't really tie in with existing patterns...